packages/dd-trace/src/span_processor.js

```diff
@@ -14,6 +15,10 @@ class SpanProcessor {
     const spanContext = span.context()
     const trace = spanContext._trace
 
+    if (this._exporter.isInCI) {
+      trace.origin = CI_APP_ORIGIN
```

Comment:
Why set this here and not earlier? For example at the tracer level.

Reply:
this is where trace spans are being formatted and exported, it seemed like a good place to do changes that affect the whole trace. Where would you place this? In the `trace` function in `packages/dd-trace/src/tracer.js` ?

Reply:
Something like that. It would also have the benefit that the trace would be marked right away, which would enable conditions based on the origin anywhere in the code without needing access to the config. I guess it could even be in the plugins if we always want any trace generated within Mocha or Jest for example to go to CI visibility instead of APM.

Reply:
updated in https://github.com/DataDog/dd-trace-js/pull/1734/commits/fa0561264ba015ea702d9bd9bff197fdea172e62. 

> I guess it could even be in the plugins if we always want any trace generated within Mocha or Jest for example to go to CI visibility instead of APM. 

I don't follow: do you mean to set an attribute of the tracer in the testing framework plugins? And read that attribute within `trace`?

Reply:
I think I need more detail about what the PR is actually trying to do. If the goal is to know when we're running in a test runner, wouldn't it be better to simply set the origin in those integrations?

Reply:
how would you set the origin in the integration part? Would you access the `tracer` [here](https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-plugin-mocha/src/index.js#L242) and add a flag or something?